packages/amplify-category-api/src/provider-utils/awscloudformation/utils/graphql-schema-utils.ts

```diff
@@ -118,3 +78,46 @@ const parseSchema = (schemaContent: string | undefined, pathToSchemaFile: string
     throw new Error(`The schema file at ${pathToSchemaFile} is not a valid GraphQL document. ${err?.message}`);
   }
 };
+
+const buildSchemaFromConnection = async (envName: string, databaseConfig: ImportedDataSourceConfig): Promise<Schema> => {
+  // Establish the connection
+  let adapter: DataSourceAdapter;
+  let schema: Schema;
+  const UNABLE_TO_CONNECT_MESSAGE =
+    'Failed to connect to the specified RDS Data Source. Check the connection details in the schema and re-try. Use "amplify api update-secrets" to update the user credentials.';
+
+  switch (databaseConfig.engine) {
+    case ImportedRDSType.MYSQL:
+      adapter = new MySQLDataSourceAdapter(databaseConfig as MySQLDataSourceConfig);
+      schema = new Schema(new Engine('MySQL'));
+      break;
+    case ImportedRDSType.POSTGRESQL:
+      adapter = new PostgresDataSourceAdapter(databaseConfig as PostgresDataSourceConfig);
+      schema = new Schema(new Engine('Postgres'));
+      break;
+    default:
+      printer.error('Only MySQL and Postgres Data Sources are supported.');
```

Comment:
Need to throw, unless I'm misunderstanding that `printer.error` throws in addition to printing the error message.